### PR TITLE
Add Speech Commands, docs and tests to NeMo

### PR DESCRIPTION
**Pull request by titu1994**
# Changelog

## Additions
- Add `kernel_width`, `se`, and `se_reduction_ratio` parameters to `JasperBlock` to support conv kernel width alteration and squeeze-and-excitation sub-modules.
- Add `JasperDecoderForClassification`
- Add `CropOrPadSpectrogramAugmentation` from OpenSeq2Seq
- Add `AudioToSpeechLabelDataLayer` and supporting collection and Dataset
- Add `CrossEntropyLossNM`
- Add `WhiteNoisePerturbation`
- Add `WarmupHoldPolicy`, `PolynomialDecayAnnealing` and `PolynomialHoldDecayAnnealing` learning rate policies
- Add logging callbacks and helpers
- Add "speech_command" section to docs
- Add `scripts/process_speech_commands_data.py` for dataset download and manifest building

## Changes
- Refactor `manifest.iter_item` to accept an optional Callable which will be used to parse each line of the manifest. When no default Callable is provided, internally will switch to use the private ASR parser within that module.
- Check `isinstance(optimizer, torch.optim.**O**ptimizer)` when passing instantiated optimizer instead of string to Factory otherwise python complains that you cannot test against a module `torch.optim.**o**ptimizer`

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -65,14 +65,14 @@ pipeline {
       parallel { 
         stage('BERT on the fly preprocessing') {
           steps {
-            sh 'cd examples/nlp/language_modeling && CUDA_VISIBLE_DEVICES=0 python bert_pretraining.py --amp_opt_level O1 --data_dir /home/mrjenkins/TestData/nlp/wikitext-2 --dataset_name wikitext-2 --work_dir outputs/bert_lm/wikitext2 --batch_size 64 --lr 0.01 --lr_policy CosineAnnealing --lr_warmup_proportion 0.05 --tokenizer sentence-piece --vocab_size 3200 --hidden_size 768 --intermediate_size 3072 --num_hidden_layers 6 --num_attention_heads 12 --hidden_act "gelu" --save_step_freq 200 --sample_size 10000000 --mask_probability 0.15 --short_seq_prob 0.1 --max_steps=300'
```
Comment (okuchaiev) [inline Jenkinsfile]: why there are changes in BERT tests for this PR?

File: nemo/collections/asr/losses.py [hunk 2]
```diff
@@ -58,3 +58,42 @@ def _loss(self, log_probs, targets, input_length, target_length):
 
     def _loss_function(self, **kwargs):
         return self._loss(*(kwargs.values()))
+
+
+class CrossEntropyLossNM(LossNM):
```
Comment (ekmb) [inline nemo/collections/asr/losses.py]: could you reuse this: https://github.com/NVIDIA/NeMo/blob/master/nemo/backends/pytorch/common/losses.py#L99

File: Jenkinsfile (hunk 1 above)
Comment (titu1994) [reply Jenkinsfile]: Will look into this, probably from a pull from master somewhere as I didn't update Jenkinsfile at all.

File: nemo/collections/asr/losses.py (hunk 2 above)
Comment (titu1994) [reply nemo/collections/asr/losses.py]: Woops. Didn't see this one before. Will revert my version.

Comment (titu1994) [reply nemo/collections/asr/losses.py]: Didn't see this before. I'll revert my version.

Comment (titu1994) [reply nemo/collections/asr/losses.py]: Since that is a backend function, should I extend the above class and just add `reduction="mean"` as an argument? Would be a much cleaner implementation.

Comment (ekmb) [reply nemo/collections/asr/losses.py]: I would just modify this one to be more flexible.

Comment (titu1994) [reply nemo/collections/asr/losses.py]: I added the reduction kwarg to the backend function and aliased it in asr collection so that it can be used as part of the collection.
